### PR TITLE
Update PageFetcherSnapshotTests using deprecated test API

### PR DESCRIPTION
**Pull request by claraf3**
Behavior of coroutine dispatchers changed when switching to
kotlinx-coroutines-test 1.6 API, requiring manual adjustments.

Test: ./gradlew paging:paging-common:test
Bug: 220884819

### CODE REVIEW

File: paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt [hunk 1]
```diff
@@ -2713,9 +2704,8 @@ class PageFetcherSnapshotTest {
         }
     }
 
-    @Suppress("DEPRECATION") // b/220884819
     @Test
-    fun remoteMediator_endOfPaginationNotReachedLoadStateAppend() = runBlockingTest {
+    fun remoteMediator_endOfPaginationNotReachedLoadStateAppend() = testScope.runTest {
```
Comment (dlam) [inline paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt]: Can we revert the change for this test for now, since we're missing an event and it seems like it'll take some work to make this test a bit more consistent?

File: paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt [hunk 2]
```diff
@@ -2505,9 +2501,8 @@ class PageFetcherSnapshotTest {
         state.job.cancel()
     }
 
-    @Suppress("DEPRECATION") // b/220884819
     @Test
-    fun remoteMediator_endOfPaginationNotReachedLoadStatePrepend() = runBlockingTest {
+    fun remoteMediator_endOfPaginationNotReachedLoadStatePrepend() = testScope.runTest {
```
Comment (dlam) [inline paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt]: Do we need to use testScope here? Let's try to get rid of it if we don't need it since it relies on UnconfinedTestDispatcher.

Comment (claraf3) [reply paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt]: This test is also missing an event like remoteMediator_endOfPaginationNotReachedLoadStateAppend(). I'll revert this one as well.

And it does seem to require UnconfinedTestDispatcher. If we use the StandardTestDispatcher, coroutines are not getting launched properly but the test setup doesn't allow more refined control of launches. I'll leave a comment on the bug regarding this.

File: paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt (hunk 1 above)
Comment (claraf3) [reply paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt]: Reverted along with its prepend counterpart remoteMediator_endOfPaginationNotReachedLoadStatePrepend

File: paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt (hunk 2 above)
Comment (claraf3) [reply paging/paging-common/src/test/kotlin/androidx/paging/PageFetcherSnapshotTest.kt]: reverted along with its Append counterpart and noted in bug report the need for new test setups to guarantee launch order

**Review by dlam — APPROVED** [review]
